### ISSUE TITLE
How to use VIT?

### ISSUE THREAD

**Issue by Elementcollector1** [marytts/marytts#772]
I'm super new to pretty much everything about this, and I can't seem to find the voice installation tools on Mary 5.2. I got the server and client to open just fine, but there's nowhere in the GUI to work with preexisting voices. I've downloaded the builder version, the regular version, and the installer, and I think I found something to do with the Voice Import Tools in the builder's kit (marytts-builder-5.2/bin/voiceimport.sh), but I can't run it anywhere.

How do I install/run the VIT?

Also, how do I create transcript files? There's something about the 'MARYXML' format, but I don't know how to create usable files in that format. Different, better formats were mentioned here (https://github.com/marytts/marytts/wiki/VoiceImportToolsTutorial), but I've found no way to create files under those formats either.

**Comment by psibre** [comment]
Please clarify what you mean by "VIT".

If you want to install provided ("preexisting"?) voices into your MaryTTS 5.2 instance, you can run the `marytts-component-installer` GUI.

We also provide the *marytts-installer* as a separate package for download, which is just a web installer and should work out of the box.

If you want to build new MaryTTS voices from your own data, you can use the legacy "VoiceImportTools", which are [explained in the wiki](../wiki/VoiceImportToolsTutorial).

Or you can try out the new [Gradle-based voicebuilding process](../../gradle-marytts-voicebuilding-plugin). It's is still experimental, and not well documented, but works just like for the various voice repositories we maintain on GitHub, e.g., [marytts/voice-cmu-slt](../../voice-cmu-slt). I also have [some slides online](http://www.coli.uni-saarland.de/~steiner/teaching/2016/summer/voicebuilding/slides/).

**Comment by Elementcollector1** [comment]
By VIT, I mean the Voice Import Tools that are supposedly included in version 5.2 according to this link (https://github.com/marytts/marytts/wiki/VoiceImportToolsTutorial). These should allow one to create a voice from a set of existing voice recordings.

That Gradle-based one looks like the one I want. How do I integrate it into the installed version of MaryTTS?

**Comment by psibre** [comment]
> supposedly included in version 5.2

It's included as the "marytts-builder" package.

> How do I integrate it into the installed version of MaryTTS?

You don't; it uses MaryTTS as a dependency.

**Comment by Elementcollector1** [comment]
I see. Well, I'm trying the 'transcription' Windows batch file, and it opens the cmd.exe window for about half a second and then closes it without doing anything else. From what I've searched, this sounds like the program is failing to run, but I can't really seem to see why. Right now it's in Desktop/marytts-builder 5.2/marytts-builder-5.2/bin/transcription.bat.

The rest of MaryTTS works fine, so I'm greatly confused as to why this isn't. I've tried running it while server and server/client for regular MaryTTS are running, with no success, and I have also tried reinstalling the JDK (8u131) to run the .jar files, with equally dismal results ('JNI error has occurred, check installation' followed by 'A Java machine exception has occurred'). Any ideas as to how to fix one or both of these?

**Comment by psibre** [comment]
Not to spoil the fun, but you will not be able to build voices under Windows, unless you can compile all required third-party tools.

Your best bet is doing everything in a Linux VM.

**Comment by Elementcollector1** [comment]
I'm aware of the third-party tools, and that I need to install them. However, I felt I should get the interface up and running first. One step at a time, you know?

**Comment by psibre** [comment]
Not sure what you mean by "interface", but if you're trying to build voices on Windows, you're probably wasting your time at this point. Please feel free to prove me wrong though.

**Comment by Elementcollector1** [comment]
Hmm. I can install Arch Linux on a Raspberry Pi that I'm planning to eventually use for this project, would that work? I've never used Linux before, so I've no idea what to expect.

By 'interface', I mean the graphical user interface (GUI). I assumed that MaryTTS would be able to at least load the initial menu without the third-party addons - is this incorrect?

Speaking of third-party addons: The ones I need are Praat, Edinburgh Speech Tools, and EHMM, right? What do I do with them when I have them, do I put them in the MaryTTS folder somewhere?

**Comment by psibre** [comment]
> Hmm. I can install Arch Linux on a Raspberry Pi that I'm planning to eventually use for this project, would that work? I've never used Linux before, so I've no idea what to expect.

Personally, I've never used Arch Linux, so I can't comment on that. But assuming you can install all required tools using `pacman` or from source, it should work.

The Pis have CPU, memory, and storage stats that are *years* behind current desktops/laptops. Even running VirtualBox in Windows will get you much better performance on a modern system.

> By 'interface', I mean the graphical user interface (GUI). I assumed that MaryTTS would be able to at least load the initial menu without the third-party addons - is this incorrect?

Why do you want a GUI where you have to click lots of checkboxes and buttons, if you can just run `./gradlew build`? But if you *really* want a GUI, you could run `./gradlew --gui` and then click on "build"... for now.

> Speaking of third-party addons: The ones I need are Praat, Edinburgh Speech Tools, and EHMM, right? What do I do with them when I have them, do I put them in the MaryTTS folder somewhere?

Interesting choice of words... Like it says in all of the voice-* READMEs, you need sox, praat, and speech-tools.

EHMM was one possible third-party software for forced alignment, but I would recommend you use something better nowadays. The [WebMAUS](https://clarin.phonetik.uni-muenchen.de/BASWebServices/#!/services/WebMAUSBasic) service works reasonably well (although you will probably have to edit the labels), and our current tool of choice is a [dockerized Kaldi wrapper](../../marytts-dockerfiles/tree/master/kaldi-mfa). It's a bit hacky at the moment, but we hope to find time to streamline that over the summer.

**TL;DR** Install Ubuntu in VirtualBox and try building [marytts/voice-cmu-slt](../../voice-cmu-slt) before you get lost in the details.
